### PR TITLE
feat(longitudinal_controller): skip integral in manual mode

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

In the current implementation, if the vehicle speed is almost zero, the control error integral is skipped. However, when the vehicle is moving under manual control, the integration is performed which is unexpected behavior. This causes an overshoot when the vehicle is engaged while driving.

This PR fixes the issue by disabling the integration when the control_mode is not in autonomous.

## Related links

must be merged with https://github.com/tier4/autoware_launch/pull/683

## Tests performed

- psim, run/stop

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/pid_longitudinal_controller/include/pid_longitudinal_controller/pid_longitudinal_controller.hpp [hunk 1]
```diff
@@ -94,6 +94,7 @@ class PidLongitudinalController : public trajectory_follower::LongitudinalContro
   nav_msgs::msg::Odometry m_current_kinematic_state;
   geometry_msgs::msg::AccelWithCovarianceStamped m_current_accel;
   autoware_auto_planning_msgs::msg::Trajectory m_trajectory;
+  autoware_auto_vehicle_msgs::msg::ControlModeReport m_current_control_mode;
```
Comment (takayuki5168) [inline control/pid_longitudinal_controller/include/pid_longitudinal_controller/pid_longitudinal_controller.hpp:97]: Let me think about whether `control mode report` is the best one to know the current state is manual or autonomous.

internal link
https://star4.slack.com/archives/C4P0NSMB5/p1673340967431229

Comment (TakaHoribe) [reply control/pid_longitudinal_controller/include/pid_longitudinal_controller/pid_longitudinal_controller.hpp]: @takayuki5168 Thank you. It is addressed in https://github.com/autowarefoundation/autoware.universe/pull/2619/commits/2f0948954fffdf65174210bc93eea5244b77da68 : `ControlModeReport` -> `OperationModeState`

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Comment by takayuki5168** [comment]
@TakaHoribe Test on CI failed. Regarding the smoke_test, operation_mode must be published so that the input_data will be initialized?

**Pull request merged**